push/src/main/java/org/aerogear/mobile/push/PushService.java

```diff
@@ -0,0 +1,250 @@
+                            + PUSH_CONFIG_FILE_NAME + ". Please check if the file exists");
+        } finally {
+            if (fileStream != null) {
+                try {
+                    fileStream.close();
+                } catch (IOException e) {
+                    // Ignore IOException
+                }
+            }
+        }
+    }
+
+    public void registerDevice(final Callback callback) {
+        registerDevice(new UnifiedPushConfig(), callback);
+    }
+
+    public void registerDevice(final UnifiedPushConfig unifiedPushConfig, final Callback callback) {
+
+        try {
+
+            JSONObject data = new JSONObject();
+
+            data.put("deviceToken", FirebaseInstanceId.getInstance().getToken());
+            data.put("deviceType", deviceType);
+            data.put("operatingSystem", operatingSystem);
+            data.put("osVersion", osVersion);
+            data.put("alias", unifiedPushConfig.getAlias());
+
+            final List<String> categories = unifiedPushConfig.getCategories();
+            if (categories != null && !categories.isEmpty()) {
+                JSONArray jsonCategories = new JSONArray();
+                for (String category : categories) {
+                    jsonCategories.put(category);
+                }
+                data.put("categories", jsonCategories);
+            }
+
+            String authHash = getHashedAuth(unifiedPushCredentials.getVariant(),
+                            unifiedPushCredentials.getSecret().toCharArray());
+
+            final HttpRequest httpRequest = core.getHttpLayer().newRequest();
+            httpRequest.addHeader("Authorization", authHash);
+            httpRequest.post(url + registryDeviceEndpoint, data.toString().getBytes());
+
+            final HttpResponse httpResponse = httpRequest.execute();
+            httpResponse.onSuccess(new Runnable() {
+                @Override
+                public void run() {
+                    switch (httpResponse.getStatus()) {
+                        case HTTP_OK:
+
+                            FirebaseMessaging firebaseMessaging = FirebaseMessaging.getInstance();
+
+                            if (categories != null) {
+                                for (String catgory : categories) {
+                                    firebaseMessaging.subscribeToTopic(catgory);
+                                }
+                            }
+
+                            firebaseMessaging.subscribeToTopic(unifiedPushCredentials.getVariant());
+
+                            callback.onSuccess();
+                            break;
+                        default:
+                            callback.onError(new HttpException());
+                            break;
+                    }
+                }
+            });
+
+        } catch (JSONException e) {
+            MobileCore.getLogger().error(e.getMessage(), e);
+            callback.onSuccess(e);
```

Comment:
Should onSuccess be called here?

Reply:
Nopz, this is a poor copy and paste :tired_face: